src/main/java/seedu/address/model/session/SessionDate.java

```diff
@@ -23,15 +24,15 @@
      * @param dateValue string of date in YYYY-MM-DD format
      * @param timeValue string of time in HH:MM format
      */
-    public SessionDate(String dateValue, String timeValue) throws SessionException {
+    public SessionDate(String dateValue, String timeValue) {
         try {
             LocalDate localDate = LocalDate.parse(dateValue);
             LocalTime localTime = LocalTime.parse(timeValue);
 
             LocalDateTime localDateTime = localDate.atTime(localTime);
             this.dateTime = localDateTime;
         } catch (DateTimeParseException e) {
-            throw new SessionException(INCORRECT_DATE_TIME_FORMAT_ERROR_MESSAGE + e, e);
+            checkArgument(false, MESSAGE_CONSTRAINTS + e.getMessage());
         }
```

Comment:
Since we have the isValidSessionDate() check now, we do not need this try and catch block anymore since it is a guarantee that we will receive a valid dateValue and timeValue in our SessionDate constructor. 
There are a few changes to other files regarding this issue as well. I will do the changes once this PR is merged.